networkx/convert_matrix.py

```diff
@@ -121,51 +121,101 @@ def to_pandas_dataframe(G, nodelist=None, multigraph_weight=sum, weight='weight'
     df = pd.DataFrame(data=M, index = nodelist ,columns = nodelist)
     return df
 
-def from_pandas_dataframe(df,create_using=None):
+def from_pandas_dataframe(df, source, target, edge_attr=None,
+        create_using=None):
     """Return a graph from Pandas DataFrame.
 
-    The Pandas DataFrame is interpreted as an adjacency matrix for the graph.
+    The Pandas DataFrame should contain at least two columns of node names and
+    zero or more columns of node attributes. Each row will be processed as one
+    edge instance.
+
+    Note: This function iterates over DataFrame.values, which is not
+    guaranteed to retain the data type across columns in the row. This is only
+    a problem if your row is entirely numeric and a mix of ints and floats. In
+    that case, all values will be returned as floats. See the
+    DataFrame.iterrows documentation for an example.
 
     Parameters
     ----------
     df : Pandas DataFrame
-      An adjacency matrix representation of a graph
+      An edge list representation of a graph
 
-    create_using : NetworkX graph
-       Use specified graph for result.  The default is Graph()
+    source : str or int
+      A valid column name (string or iteger) for the source nodes (for the
+      directed case).
 
-    Notes
-    -----
-    If the numpy matrix has a single data type for each matrix entry it
-    will be converted to an appropriate Python data type.
+    target : str or int
+      A valid column name (string or iteger) for the target nodes (for the
```

Comment:
Indentation for the parameter description should be 4 spaces in (according to the NumPy docstring standard, which we follow). So if you could update all the descriptions in each docstring---it looks like you need to add 2 more spaces.